internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -739,12 +739,30 @@ func (mc *mac) createMistStream(streamName string, stream *livepeer.CreateStream
 	return err
 }
 
+func (mc *mac) handleHealthcheck(w http.ResponseWriter, r *http.Request) {
+	if !mc.isEtcdSessionHealthy() {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (mc *mac) isEtcdSessionHealthy() bool {
+	select {
+	case <-mc.etcdSession.Done():
```

Comment:
this will crash if mapic used without ETCD

Reply:
Fixed.

Turns out that the background `recoverSessionLoop` was also broken, and would panic mapic just after starting. I didn't know that running it without an etcd was actually still a thing. Do we have any of those happening in produciton right now? (cause they're likely to be broken due to the recover loop)

Reply:
> Do we have any of those happening in produciton right now? (cause they're likely to be broken due to the recover loop)

No, but ETCD is an still optional in mapic, so we should keep it working without it - or make ETCD mandatory.